docs/app/Components/ComponentDoc/ComponentProps/ComponentProps.js

```diff
@@ -38,123 +36,22 @@ export default class ComponentProps extends Component {
-
-    const tagDescriptionRows = _.compact([...params, returns]).map((tag) => {
-      const name = tag.name || tag.title
-      return (
-        <div key={name} style={{ display: 'flex', flexDirection: 'row' }}>
-          <div style={{ flex: '2 2 0', padding: '0.1em 0' }}>
-            <code>{name}</code>
-          </div>
-          <div style={{ flex: '5 5 0', padding: '0.1em 0' }}>
-            {tag.description}
-          </div>
-        </div>
-      )
-    })
-
-    return (
-      <ComponentPropsExtra title={<pre>{item.name}({paramSignature}){returns ? `: ${getTagType(returns)}` : ''}</pre>}>
-        {tagDescriptionRows}
-      </ComponentPropsExtra>
-    )
-  }
-
-  renderEnums = ({ name, type, value }) => {
-    const { showEnumsFor } = this.state
-
-    if (type !== '{enum}' || !value) return
-    return (
-      <ComponentPropsEnum
-        showAll={showEnumsFor[name]}
-        toggle={this.toggleEnumsFor(name)}
-        values={value}
-      />
-    )
-  }
-
-  renderRow = item => (
-    <Table.Row key={item.name}>
-      <Table.Cell collapsing>{this.renderName(item)}{this.renderRequired(item)}</Table.Cell>
-      <Table.Cell collapsing>{this.renderDefaultValue(item)}</Table.Cell>
-      <Table.Cell collapsing>{item.type}</Table.Cell>
-      <Table.Cell>
-        {item.description && <p>{item.description}</p>}
-        {this.renderFunctionSignature(item)}
-        {this.renderEnums(item)}
-      </Table.Cell>
-    </Table.Row>
-  )
-
   render() {
     const { props: propsDefinition } = this.props
-
-    const content = _.sortBy(_.map(propsDefinition, (config, name) => {
-      const value = _.get(config, 'type.value')
-      let type = _.get(config, 'type.name')
-      if (type === 'union') {
-        type = _.map(value, val => val.name).join('|')
-      }
-      type = type && `{${type}}`
-
-      const description = _.get(config, 'docBlock.description', '')
-
-      return {
-        name,
-        type,
-        value,
-        tags: _.get(config, 'docBlock.tags'),
-        required: config.required,
-        defaultValue: config.defaultValue,
-        description: description && description.split('\n').map(l => ([l, <br key={l} />])),
-      }
-    }), 'name')
+    const { showEnumsFor } = this.state
 
     return (
       <Table compact='very' basic='very'>
-        <Table.Header>
-          <Table.Row>
-            <Table.HeaderCell>Name</Table.HeaderCell>
-            <Table.HeaderCell>Default</Table.HeaderCell>
-            <Table.HeaderCell>Type</Table.HeaderCell>
-            <Table.HeaderCell>Description</Table.HeaderCell>
-          </Table.Row>
-        </Table.Header>
+        <ComponentPropsHeader />
         <Table.Body>
-          {_.map(content, this.renderRow)}
+          {_.map(propsDefinition, item => (
+            <ComponentPropsRow
+              {...item}
+              key={item.name}
+              showEnums={showEnumsFor[item.name]}
+              onToggleEnum={this.toggleEnumsFor(item.name)}
```

Comment:
Since the `showEnums` state is only needed within `ComponentPropsRow`, I think we can push it down another level.  The parent doesn't need to re-render when the child shows/hides enums.